### PR TITLE
update url for kubernetes-dev mailing list to new location

### PR DESCRIPTION
**Pull request by csantanapr**
#### What type of PR is this:
/kind documentation

#### What this PR does / why we need it:
Is better to have direct link to the new location of mailing list for dev@kubernetes.io

#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:
This was s recent change, the mailing list change location

/assign @AuraSinis

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/lead.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ hold and only released once approved by a SIG Release (subproject) owner.**
   - [ ] Has joined the following mailing lists:
     - [kubernetes-sig-release](https://groups.google.com/forum/#!forum/kubernetes-sig-release)
     - [kubernetes-release-team](https://groups.google.com/a/kubernetes.io/g/release-team)
-    - [kubernetes-dev](https://groups.google.com/forum/#!forum/kubernetes-dev)
+    - [kubernetes-dev](https://groups.google.com/a/kubernetes.io/group/dev)
```
Comment (saschagrunert) [inline .github/ISSUE_TEMPLATE/lead.md:27]: nit:
```suggestion
    - [kubernetes-dev](https://groups.google.com/a/kubernetes.io/g/dev)
```

Comment (csantanapr) [reply .github/ISSUE_TEMPLATE/lead.md:27]: @saschagrunert I did a find and replace on the repo to replace the url.
The url (`…/goup/dev`) that I used is the one that is in the old mailing list telling users with a message and this url.

It looks like both are correct `…/group/dev` and `…/g/dev` the first one redirects to the second one with the `g`.

**Comment by csantanapr** [comment]
Thank you @saschagrunert for the PR review 🙏  I addressed your comments. 👍

File: leads/onboarding.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ To ensure quality communication SIG Release Chairs and Technical Leads should:
   - kubernetes-release-team:
     https://groups.google.com/a/kubernetes.io/g/release-team
   - kubernetes-dev:
-    https://groups.google.com/forum/#!forum/kubernetes-dev
+    https://groups.google.com/a/kubernetes.io/group/dev
```
Comment (cpanato) [inline leads/onboarding.md:18]: maybe can apply the same here `https://groups.google.com/a/kubernetes.io/g/dev`

File: release-engineering/role-handbooks/branch-manager.md [hunk 3]
```diff
@@ -185,7 +185,7 @@ krel announce send --tag vX.Y.0-{alpha,beta,rc}.Z --name "<Your Name>" --name <Y
 See the [Release Commands Cheat Sheet](https://github.com/kubernetes/sig-release/blob/master/release-engineering/role-handbooks/patch-release-team.md#release-commands-cheat-sheet) for example commands.
 
 [k-announce-list]: https://groups.google.com/forum/#!forum/kubernetes-announce
-[k-dev-list]: https://groups.google.com/forum/#!forum/kubernetes-dev
+[k-dev-list]: https://groups.google.com/a/kubernetes.io/group/dev
```
Comment (cpanato) [inline release-engineering/role-handbooks/branch-manager.md:188]: and here as well

File: release-team/role-handbooks/docs/Release-Timeline.md [hunk 4]
```diff
@@ -155,7 +155,7 @@ Send a Slack message to those that you select, e.g:
 You'll then want to join these mailing lists if you haven't already:
 >- https://groups.google.com/forum/#!forum/kubernetes-sig-release
 >- https://groups.google.com/forum/#!forum/kubernetes-sig-docs
->- https://groups.google.com/forum/#!forum/kubernetes-dev
+>- https://groups.google.com/a/kubernetes.io/group/dev
```
Comment (cpanato) [inline release-team/role-handbooks/docs/Release-Timeline.md:158]: one more :)

File: leads/onboarding.md (hunk 2 above)
Comment (csantanapr) [reply leads/onboarding.md:18]: Let me check again, I swear I did a find and replace all.

Comment (csantanapr) [reply leads/onboarding.md]: Good catch @cpanato those where not markdown links

**Review by cpanato — APPROVED** [review]
thanks!!
/lgtm

**Pull request merged**